### DISCUSSION TITLE
How to use showProperties with Relations plugin?

### DISCUSSION THREAD

**Discussion by sjorobekov** [SoftwareBrothers/admin-bro#1675] [Q&A]
Hello,

I'm using AdminJS as AdonisJS module with Relations plugin.
I use following config:

```typescript
export const createOrganizationResource = () => ({
  resource: new LucidResource(Organization, 'postgres'),
  options: {
    listProperties: ['name', 'active'],
    editProperties: [
      'name',
      'active',
    ],
    showProperties: [
      'name',
      'active',
      'address',
    ],
  },
  features: [
    owningRelationSettingsFeature({
      componentLoader,
      licenseKey: process.env.LICENSE_KEY!,

      relations: {
        users: {
          type: RelationType.OneToMany,
          target: {
            joinKey: 'organizationId',
            resourceId: 'users',
          },
        },
      },
    }),
  ],
})
```

This config hides Users list under Organization Show page. If I don't use showProperties, it shows Users list. How to avoid this behavior?

**Comment by mohamad-zbib** [comment]
first make sure the you are using eager loading for relationships.

then build a custom react components and use it like:

    application: {
      components: {
        list: Components.ApplicationNameList,
        show: Components.ApplicationNameShow,
      },
    },
    progress: {
      isVisible: { filter: false, list: true, show: true },
      components: {
        list: Components.RequestProgressList,
        show: Components.RequestProgressShow,
      },
    },

**Comment by dziraf** [comment]
Make sure you include either `relations` or `users` in `showProperties` (I don't remember if `@adminjs/relations` uses a fixed name or relationship's name and cannot check right now)

**Reply by evanhammer** [reply]
It's `relations`
